### ISSUE TITLE
Suggestion: throwing counterpart to `fail(message`

### ISSUE THREAD

**Issue by noamfreeman** [Quick/Nimble#1161]
a common pattern in swift tests is failing after checking some condition in a if/guard
```swift
guard let case .someCase(associatedValue) = someEnum else {
  fail("some explanation")
  return
}
```

it could be convenient to have a `require` counterpart to the `fail` function that throws an error.

Of coarse, users could always throw their own errors, but for users aren't used to doing so, it involves a lot of considerations.

it is possible to achieve this behavior with the existing `require` DSL, but it looks a bit wierd:
```swift
try require(false).verify(false, FailureMessage(stringValue: description), nil)
```
or as a utility:
```swift
private func failWithError(
  file: FileString = #filePath, line: UInt = #line,
  _ description: String
) throws -> Never {
  try require(file: file, line: line, false).verify(false, FailureMessage(stringValue: description),
                                                    nil)
  throw SomeErrorSayingWeCantGetHere
}
```

this would be simplified greatly with the access to the `RequireError` that does nice things.
it could be simple as adding
```swift
public func throwingFail(_ message: String, location: SourceLocation) throws -> Never {
    let handler = NimbleEnvironment.activeInstance.assertionHandler
    handler.assert(false, message: FailureMessage(stringValue: message), location: location)
    throw RequireError(message: message, location: location)
}
```
and the `,file, line)` counterparts to `DSL+Require`

sorry in advance if there is already something simpler that could be used as a throwing `fail`

**Comment by younata** [comment]
Thanks for the suggestion! This is a really good idea!

**Comment by younata** [comment]
Just merged https://github.com/Quick/Nimble/pull/1163, and it'll be released whenever #1157 is fixed.

**Comment by noamfreeman** [comment]
Thanks! that was fast!

**Comment by younata** [comment]
> Thanks! that was fast!

You're welcome! This happened to have been suggested shortly before I found time to actually implement it.

BTW, I just released Nimble 13.6.0, which includes this.
